frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -88,22 +90,36 @@ const UpdatingMessage: React.SFC<CVStatusMessageProps> = ({cv}) => {
   </React.Fragment>;
 };
 
-const ErrorRetrievingMessage: React.SFC<{}> = () => <div>
-  <RedExclamationCircleIcon /> Error retrieving
-</div>;
+const ErrorRetrievingMessage: React.SFC<CVStatusMessageProps> = ({cv}) => {
+  const retrievedUpdatesCondition = getClusterVersionCondition(cv, ClusterVersionConditionType.RetrievedUpdates, K8sResourceConditionStatus.False);
+  return (
+    <Tooltip content={truncateMiddle(retrievedUpdatesCondition.message, { length: 256 })}>
+      <span><RedExclamationCircleIcon /> Error retrieving</span>
+    </Tooltip>
+  );
+};
 
-const FailingMessage: React.SFC<{}> = () => <React.Fragment>
-  <div>
-    <RedExclamationCircleIcon /> Failing
-  </div>
-  <Link to="/settings/cluster/clusteroperators">View details</Link>
-</React.Fragment>;
+const FailingMessage: React.SFC<CVStatusMessageProps> = ({cv}) => {
+  const failingCondition = getClusterVersionCondition(cv, ClusterVersionConditionType.Failing, K8sResourceConditionStatus.True);
+  return <React.Fragment>
+    <div>
+      <Tooltip content={truncateMiddle(failingCondition.message, { length: 256 })}>
+        <span><RedExclamationCircleIcon /> Failing</span>
+      </Tooltip>
+    </div>
+    <Link to="/settings/cluster/clusteroperators">View details</Link>
+  </React.Fragment>;
+};
 
 const UpToDateMessage: React.SFC<{}> = () => <span>
   <GreenCheckCircleIcon /> Up to date
 </span>;
 
 export const UpdateStatus: React.SFC<UpdateStatusProps> = ({cv}) => {
+  if (cv.spec.channel === 'NoChannel') {
```

Comment:
@wking is `NoChannel` a thing? i can't find any references to it anywhere

Reply:
It's [a condition][1], not part of the `spec`.

[1]: https://github.com/openshift/cluster-version-operator/blob/59601a89bb7a65cf2fbaa4697d5142091f68534f/pkg/cvo/availableupdates.go#L146

Reply:
Thanks, I've implemented this incorrectly then. Let me fix :)